apps/vs-code-designer/src/app/commands/createNewCodeProject/createNewCodeProject.ts

```diff
@@ -0,0 +1,93 @@
+import { FolderListStep } from '../createNewProject/createProjectSteps/FolderListStep';
+import { OpenFolderStepCodeProject } from './CodeProjectBase/OpenFolderStepCodeProject';
+import { NewCodeProjectTypeStep } from './createCodeProjectSteps/NewCodeProjectTypeStep';
+import { setWorkspaceName } from './createCodeProjectSteps/SetWorkspaceName';
+import { setMethodName } from './createCodeProjectSteps/createFunction/setMethodName';
+import { setNameSpace } from './createCodeProjectSteps/createFunction/setNamepSpace';
+import { isString } from '@microsoft/utils-logic-apps';
+import { AzureWizard } from '@microsoft/vscode-azext-utils';
+import type { IActionContext } from '@microsoft/vscode-azext-utils';
+import { latestGAVersion, OpenBehavior } from '@microsoft/vscode-extension';
+import type { ICreateFunctionOptions, IFunctionWizardContext, ProjectLanguage, ProjectVersion } from '@microsoft/vscode-extension';
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { window } from 'vscode';
+
+export async function createNewCodeProjectFromCommand(
+  context: IActionContext,
+  folderPath?: string | undefined,
+  language?: ProjectLanguage,
+  version?: ProjectVersion,
+  openFolder = true,
+  templateId?: string,
+  functionName?: string,
+  functionSettings?: { [key: string]: string | undefined }
+): Promise<void> {
+  await createNewCodeProjectInternal(context, {
+    folderPath: isString(folderPath) ? folderPath : undefined,
+    templateId,
+    functionName,
+    functionSettings,
+    suppressOpenFolder: !openFolder,
+    language,
+    version,
+  });
+}
+
+export async function createNewCodeProjectInternal(context: IActionContext, options: ICreateFunctionOptions): Promise<void> {
+  addLocalFuncTelemetry(context);
+
+  const language: ProjectLanguage | undefined = (options.language as ProjectLanguage) || getGlobalSetting(projectLanguageSetting);
+  const version: string = options.version || getGlobalSetting(funcVersionSetting) || (await tryGetLocalFuncVersion()) || latestGAVersion;
+  const projectTemplateKey: string | undefined = getGlobalSetting(projectTemplateKeySetting);
+  const wizardContext: Partial<IFunctionWizardContext> & IActionContext = Object.assign(context, options, {
+    language,
+    version: tryParseFuncVersion(version),
+    projectTemplateKey,
+  });
+
+  if (options.folderPath) {
+    new FolderListStep.setProjectPath(wizardContext, options.folderPath);
+  }
+
+  if (options.suppressOpenFolder) {
+    wizardContext.openBehavior = OpenBehavior.dontOpen;
+  } else if (!wizardContext.openBehavior) {
+    wizardContext.openBehavior = getWorkspaceSetting(projectOpenBehaviorSetting);
+    context.telemetry.properties.openBehaviorFromSetting = String(!!wizardContext.openBehavior);
+  }
+
+  const wizard: AzureWizard<IFunctionWizardContext> = new AzureWizard(wizardContext, {
+    title: localize('createNewCodeProject', 'Create new custom logic app workspace'),
```

Comment:
@samikay101: Can you provide more context about what "custom" means here?

Reply:
> @samikay101: Can you provide more context about what "custom" means here?

@ecfan essentially this feature allows the user to create a "custom" function or code in C# that they can then call from the logic app.

Reply:
@samikay101: Thanks for the context. Unless the customer knows what "custom" means in this context, I'd suggest not including the word. The word placement is ambiguous about whether "custom" applies to "logic app" or to "workspace". The following is a clearer version:

```suggestion
    title: localize('createNewCodeProject', 'Create workspace for custom logic app'),
```